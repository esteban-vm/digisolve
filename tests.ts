import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
