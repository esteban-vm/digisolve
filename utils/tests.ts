import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
