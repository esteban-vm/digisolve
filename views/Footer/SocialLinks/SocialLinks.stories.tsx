import type { Meta } from '@storybook/react'
import { styled } from '@/styles'
import SocialLinks from './SocialLinks'

const Wrapper = styled.div`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);

  ul {
    list-style: none;

    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }
  }
`

export default {
  component: SocialLinks,
  title: 'Views/Footer/SocialLinks',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper className='row'>
          <div className='col span_1_of_2' />
          <div className='col span_1_of_2'>
            <Story {...context} />
          </div>
        </Wrapper>
      )
    },
  ],
} as Meta<typeof SocialLinks>

export const SocialLinksExample = () => <SocialLinks />
