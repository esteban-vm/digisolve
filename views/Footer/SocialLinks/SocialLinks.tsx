import type { BasicComponent } from '@/types'
import type { IconProp } from '@fortawesome/fontawesome-svg-core'
import Link from 'next/link'
import { faFacebookF, faTwitter, faGooglePlusG, faInstagram, faYoutube } from '@fortawesome/free-brands-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { styled } from '@/styles'

export const socialLinks = [faFacebookF, faTwitter, faGooglePlusG, faInstagram, faYoutube] as IconProp[]

const SocialLinksComponent: BasicComponent = (props) => {
  return (
    <ul {...props}>
      {socialLinks.map((icon, index) => (
        <li key={index}>
          <Link href='#' target='_blank'>
            <FontAwesomeIcon icon={icon} />
          </Link>
        </li>
      ))}
    </ul>
  )
}

const SocialLinks = styled(SocialLinksComponent)`
  float: right;

  li {
    margin-left: 20px;

    :hover {
      :first-of-type {
        a {
          color: var(--color-facebook);
        }
      }

      :nth-of-type(2) {
        a {
          color: var(--color-twitter);
        }
      }

      :nth-of-type(3) {
        a {
          color: var(--color-google-plus);
        }
      }

      :nth-of-type(4) {
        a {
          color: var(--color-instagram);
        }
      }

      :last-of-type {
        a {
          color: var(--color-youtube);
        }
      }
    }
  }
`

export default SocialLinks
