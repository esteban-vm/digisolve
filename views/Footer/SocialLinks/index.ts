export { default, socialLinks } from './SocialLinks'
