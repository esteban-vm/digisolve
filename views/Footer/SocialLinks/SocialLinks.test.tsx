import { render, cleanup, screen, create } from '@/tests'
import SocialLinks, { socialLinks } from './SocialLinks'

describe('ðŸ§ª SOCIAL LINKS', () => {
  describe('display tests:', () => {
    it('should display the social links', () => {
      render(<SocialLinks />)
      const parent = screen.getByRole('list')

      const items = screen.getAllByRole('listitem')
      const links = screen.getAllByRole('link')
      const icons = screen.getAllByRole('img', { hidden: true })

      expect(socialLinks.length).toBe(items.length)
      expect(socialLinks.length).toBe(links.length)
      expect(socialLinks.length).toBe(icons.length)

      for (let index = 0; index < socialLinks.length; index++) {
        const item = items[index]
        const link = links[index]
        const icon = icons[index]
        expect(link).toContainElement(icon)
        expect(item).toContainElement(link)
        expect(parent).toContainElement(item)
      }

      cleanup()
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<SocialLinks />)
      expect(tree).toMatchSnapshot()
    })
  })
})
