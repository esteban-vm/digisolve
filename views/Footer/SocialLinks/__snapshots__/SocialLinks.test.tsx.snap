// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª SOCIAL LINKS style tests: should render with correct styles 1`] = `
.emotion-0 {
  float: right;
}

.emotion-0 li {
  margin-left: 20px;
}

.emotion-0 li:hover:first-of-type a {
  color: var(--color-facebook);
}

.emotion-0 li:hover:nth-of-type(2) a {
  color: var(--color-twitter);
}

.emotion-0 li:hover:nth-of-type(3) a {
  color: var(--color-google-plus);
}

.emotion-0 li:hover:nth-of-type(4) a {
  color: var(--color-instagram);
}

.emotion-0 li:hover:last-of-type a {
  color: var(--color-youtube);
}

<ul
  className="emotion-0"
>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-facebook-f "
        data-icon="facebook-f"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-twitter "
        data-icon="twitter"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-google-plus-g "
        data-icon="google-plus-g"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M386.061 228.496c1.834 9.692 3.143 19.384 3.143 31.956C389.204 370.205 315.599 448 204.8 448c-106.084 0-192-85.915-192-192s85.916-192 192-192c51.864 0 95.083 18.859 128.611 50.292l-52.126 50.03c-14.145-13.621-39.028-29.599-76.485-29.599-65.484 0-118.92 54.221-118.92 121.277 0 67.056 53.436 121.277 118.92 121.277 75.961 0 104.513-54.745 108.965-82.773H204.8v-66.009h181.261zm185.406 6.437V179.2h-56.001v55.733h-55.733v56.001h55.733v55.733h56.001v-55.733H627.2v-56.001h-55.733z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-instagram "
        data-icon="instagram"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-youtube "
        data-icon="youtube"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </a>
  </li>
</ul>
`;
